src/tools/dart-compile.md

```diff
@@ -0,0 +1,257 @@
+---
+title: dart compile
+description: Command-line tool for compiling Dart source code.
+---
+
+Use the `dart compile` command to compile
+a Dart program to a [target platform](/platforms).
+The output — which you specify using a subcommand —
+can either include a Dart runtime or be a _snapshot_.
+
+Here's an example of using the `exe` subcommand
+to produce a self-contained executable file (`myapp.exe`):
+
+```terminal
+$ dart compile exe bin/myapp.dart
+Generated: /Users/me/myapp/bin/myapp.exe
+```
+
+The next example uses the `aot-snapshot` subcommand to
+produce a snapshot (`myapp.aot`).
+It then uses the [`dartaotruntime` command][] (which provides a Dart runtime)
+to run the snapshot:
+
+```terminal
+$ dart compile aot-snapshot bin/myapp.dart
+Generated: /Users/me/myapp/bin/myapp.aot
+$ dartaotruntime bin/myapp.aot
+```
+
+[`dartaotruntime` command]: /tools/dartaotruntime
+
+To specify the path to the output file,
+use the `-o` or `--output` option:
+
+```terminal
+$ dart compile exe bin/myapp.dart -o bin/runme
+```
+
+For more options and usage information,
+run `dart compile [<subcommand>] --help`:
+
+```terminal
+$ dart compile exe --help
+```
+
+The `dart compile` command replaces the
+`dart2native`, `dart2aot`, and `dart2js` commands.
+
+{{site.alert.note}}
+  You don't need to compile Dart programs before running them.
+  The Dart VM has a JIT (just-in-time) compiler
+  that's especially useful during development.
+  For more information on AOT (ahead-of-time) and JIT compilation,
+  see the [Platforms page](/platforms).
+{{site.alert.end}}
+
+
+## Types of output
+
+The following table shows the subcommands of `dart compile`.
```

Comment:
Just a thought (not entirely sure), but I wonder if it might be better to list this in a logical order, and not alphabetically (like the tool does, as that is how the help UI works). I think logically I'd sequence them like this:

  - exe
  - aot-snapshot
  - jit-snapshot
  - kernel
  - js

WDYT?

Reply:
sgtm